087-external-certificate-manager.md

```diff
@@ -0,0 +1,299 @@
+>
+> Fig 2: Existing and proposed workflow when the user provides a new cluster CA public cert
+
+### User Operator/Clients CA
+
+When a new Kafka cluster is created Strimzi will copy the CA public cert from the Secret identified in `clientsCa.publicCert` to the `<CLUSTER_NAME>-clients-ca-cert` Secret.
+Strimzi will add two annotations to the `<CLUSTER_NAME>-cluster-ca-cert` Secret:
+* `strimzi.io/ca-cert-hash` with the value being a hash of the certificate.
+* `strimzi.io/ca-cert-generation` initially set to 0.
+
+#### Issuing KafkaUser certificates
+
+The User operator will be updated to have four environment variables to configure the CA issuer:
+- `STRIMZI_CA_TYPE` with either the value `strimzi.io` or `cert-manager.io`
+- `STRIMZI_CM_ISSUER_NAME`
+- `STRIMZI_CM_ISSUER_KIND`
+- `STRIMZI_CM_ISSUER_GROUP`
+
+These will be set by the Cluster operator when the User operator is deployed as part of the Entity operator.
+
+When a `KafkaUser` with `spec.authentication.type` set to `tls` is creates Strimzi  will create a `Certificate` custom resource.
+Strimzi will specify the required CN/SANS in the `Certificate` resource for the user certificate.
+Strimzi will specify the `Secret` for the certificate to be stored to match the name of the `KafkaUser` resource.
+Strimzi will request the certificate in both PEM and PKCS12 format.
+Strimzi will wait for the usual operation timeout during the reconciliation loop for the `Certificate` status to indicate that the certificate has been issued before continuing.
+
+#### Tracking changes to user certificates
+
+Cert-manager will be responsible for renewing all user certificates.
+When a certificate is renewed cert-manager will update the related Secret.
+Since the user's clients are directly using the cert-manager created Secret, Strimzi will take no action.
+
+#### Handling CA cert renewals and key replacements
+
+During a reconciliation Strimzi will check the hash of the certificate stored in the user's CA public cert Secret to see if an update is needed.
+If the certificate has changed Strimzi will copy over the new certificate to replace the existing one.
+It will also update the `strimzi.io/ca-cert-hash` and increment the `strimzi.io/clients-ca-cert-generation` annotations.
+
+Once the annotations are updated Strimzi will update the annotation on the Kafka brokers Secret and the Kafka pods and roll the Kafka pods to trust the new CA cert.
+
+> ![Renewing/replacing the clients CA public cert/private key](./images/087-existing-renew-replace-clientca-certs.png)
+>
+> Fig 4: Existing workflow when the user provides a new clients CA public cert
+
+> ![Renewing/replacing the clients CA public cert/private key](./images/087-new-renew-replace-clientsca-cert.png)
+>
+> Fig 5: Proposed workflow when the user provides a new clients CA public cert
+
+## Affected/not affected projects
+
+This affects the Cluster Operator and User Operator.
+
+## Compatibility
+
+This feature will be optional and disabled by default.
```

Comment:
Does it mean we are going to put this feature behind a feature gate? If yes, should we state it clearly and mentioning it?

Reply:
From my understanding it won't be behind FG, but the default configuration will use Strimzi CA generation and users will need to explicitly configure cert-manager to use it.

Reply:
Ok I mislead what "optional and disabled by default" means here, maybe we should highlight that if nothing is set, the "type" will be implicitly considered as the strimzi.io one?